apis/python/src/tiledbsc/annotation_dataframe.py

```diff
@@ -186,14 +187,29 @@ def query(
             return self._ascii_to_unicode_dataframe_readback(slice_df)
 
     # ----------------------------------------------------------------
+    def _ascii_to_unicode_series_readback(self, column_number: int, series: pd.Series):
+        if len(series) > 0 and type(series[0]) == bytes:
+            return (column_number, True, series.map(lambda e: e.decode()))
+        else:
+            return (column_number, False, None)
+
     def _ascii_to_unicode_dataframe_readback(self, df: pd.DataFrame) -> pd.DataFrame:
         """
         Implements the 'decode on read' partof our logic as noted in `dim_select()`.
         """
-        for k in df:
-            dfk = df[k]
-            if len(dfk) > 0 and type(dfk[0]) == bytes:
-                df[k] = dfk.map(lambda e: e.decode())
+        futures = []
+        with ThreadPoolExecutor() as executor:
+            for k in df:
+                future = executor.submit(
+                    self._ascii_to_unicode_series_readback, k, df[k]
```

Comment:
Is this a pure python function? Does `dfk.map` hold the GIL?

Reply:
@Shelnutt2 I have much to learn -- will try to find out

Reply:
I did put in timings & this went from 1.7 seconds to 0.8 on a particular array